### PR TITLE
Updates to openstack terraform deploy

### PR DESCRIPTION
**Pull request by apearson-ibm**
Install via non-root user and general cleanup

* Put all user config into variables
* Remove openstack requirement for network-id
* Configure ICP to install via a non-root user (via sudo)
* Install docker-ce instead of docker.io per ICP req.

This is a cleanup of closed pull request #34  

@jwcroppe Please review...

### CODE REVIEW

File: docs/deploy-openstack-terraform.md [hunk 1]
```diff
@@ -35,27 +34,26 @@ Instructions
...
+    variables within the **variables.tf** file to point to appropriate ICP
+    architecture, edition, and tar ball.
+* Run [**terraform apply**] to start the ICP deployment
 * Sit back and relax... within about 30-40 minutes, you should be able to
   access your ICP cluster at https://<ICP_MASTER_IP_ADDRESS>:8443
 * If you're using this for anything beyond a proof-of-concept, please also take
-  the added step of setting *insecure=false* and set the *cacert* option to the
-  contents of the OpenStack certificate
+  the added step of setting the **insecure=false** variable in the **main.tf**
+  openstack provider, and the OS_CACERT environment variable.
```
Comment (jwcroppe) [inline docs/deploy-openstack-terraform.md]: s/openstack/OpenStack

File: terraform/openstack/bootstrap_icp_master.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+# Make sure we're not running some old version of docker
+/usr/bin/apt-get --assume-yes purge docker
+/usr/bin/apt-get --assume-yes purge docker-engine
+/usr/bin/apt-get --assume-yes purge docker.io
+/usr/bin/apt-get --assume-yes install \
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    software-properties-common
+# Add docker GPG key
```
Comment (jwcroppe) [inline terraform/openstack/bootstrap_icp_master.sh]: s/docker/Docker

File: terraform/openstack/bootstrap_icp_worker.sh [hunk 3]
```diff
@@ -24,10 +24,25 @@
...
+# Make sure we're not running some old version of docker
+/usr/bin/apt-get --assume-yes purge docker
+/usr/bin/apt-get --assume-yes purge docker-engine
+/usr/bin/apt-get --assume-yes purge docker.io
+/usr/bin/apt-get --assume-yes install \
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    software-properties-common
+# Add docker GPG key
```
Comment (jwcroppe) [inline terraform/openstack/bootstrap_icp_worker.sh]: Ditto... s/docker/Docker

File: terraform/openstack/variables.tf [hunk 4]
```diff
@@ -53,22 +53,42 @@ variable "openstack_flavor_id_worker_node" {
...
 
+variable "icp_install_user" {
+    description = "The user with sudo access across nodes (users section of cloud-init)"
+    default = "ubuntu"
+}
+
+variable "icp_install_user_password" {
+   description = "Password for sudo access (leave empty if using passwordless sudo access)"
+   default = ""
+}
```
Comment (jwcroppe) [inline terraform/openstack/variables.tf]: For consistency, we should add newlines after each variable name.

File: terraform/openstack/variables.tf [hunk 5]
```diff
@@ -53,22 +53,42 @@ variable "openstack_flavor_id_worker_node" {
...
 variable "icp_num_workers" {
     description = "The number of ICP worker nodes to provision."
```
Comment (jwcroppe) [inline terraform/openstack/variables.tf]: For consistency, let's remove the periods at the end of the descriptions

**Comment by apearson-ibm** [comment]
@jwcroppe All comments above addressed...

**Comment by jwcroppe** [comment]
@apearson-ibm @paulczar @tpouyer Changes LGTM!

**Pull request merged**